### PR TITLE
Implementation of TLV2544

### LINKED ISSUE #36: Implement TLV2544ID ADC

### PR DESCRIPTION
**Pull request by CaiB**
Resolves #36.

It doesn't have the full functionality of the device, but the rest of the features largely don't make sense for use on a non-realtime OS anyways. Appears to work reliably in my testing, once you put the required capacitor on there ;)

### CODE REVIEW

**Comment by CaiB** [comment]
Put this on hold for now. There appears to be long-term instability in my implementation.

**Comment by CaiB** [comment]
I believe I have fixed the remaining issues, and the ADC looks to be working well now. I have re-added sample clock source selection, since using the internal oscillator requires only 4-8us for sample time, and even adjacent SPI commands on the Pi take significantly longer than this, so I don't expect a timing issue. That said, there is technically no guarantee that there will be enough delay for sampling to finish.

One of the problems was that multi threaded SPI access on the Pi was causing data corruption, which needs to be fixed in the RPi SPI implementation.

File: Components/Inputs/TLV2544.cs [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+            return (ushort)(Command == Command.READ_CONF ?
+                (((DataIn[0] & 0b0000_1111) << 8) | (DataIn[1])) :
+                (DataIn[0] << 4) | ((DataIn[1] & 0b1111_0000) >> 4));
+        }
+
+        #region Structs and Enums
+        public struct Configuration
+        {
+            public VoltageReference VoltageRef;
+            public bool UseLongSample; // TODO: Check if long sampling is working.
```
Comment (Baldstrom) [inline Components/Inputs/TLV2544.cs]: Have you checked if long sampling is working?

File: Components/Inputs/TLV2544.cs [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+                    case ConversionClockSrc.SCLK:
+                        DataOut = (this.Config.UseLongSample ? new byte[2 + 4] : new byte[2 + 2]);
+                        break;
+                    case ConversionClockSrc.SCLK_HALF:
+                        DataOut = (this.Config.UseLongSample ? new byte[2 + 7] : new byte[2 + 4]);
+                        break;
+                    case ConversionClockSrc.SCLK_QUARTER:
+                        DataOut = (this.Config.UseLongSample ? new byte[2 + 14] : new byte[2 + 7]);
+                        break;
+                    case ConversionClockSrc.INTERNAL: // We are assuming the time between The two SPI transactions is at least 4us for short, or 8us for long sampling. This seems to always be true on the Pi, but is not guaranteed.
```
Comment (Baldstrom) [inline Components/Inputs/TLV2544.cs]: This comment might be important enough to warrant going in the method comment or class comment, even though I doubt people will have problems with this.

File: Components/Inputs/TLV2544.cs [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+            ConversionMode = ConversionMode.SINGLE_SHOT,
+            UseEOCPin = false,
+            FIFOTriggerLevel = FIFOTrigger.FIFO_8b
+        };
+        public readonly AnalogueInTLV254x[] Inputs;
+
+        private readonly ISPIBus Bus;
+        private readonly IDigitalOut CS;
+        private Configuration Config = DefaultConfig;
+        private sbyte ReuseChannel = -1; // TODO: Implement channel re-use to speed up repeated channel reads.
```
Comment (Baldstrom) [inline Components/Inputs/TLV2544.cs]: Are we going to wait on this feature?

File: Components/Inputs/TLV2544.cs (hunk 1 above)
Comment (CaiB) [reply Components/Inputs/TLV2544.cs]: I have not, I'll give it a try tonight before release.

File: Components/Inputs/TLV2544.cs (hunk 2 above)
Comment (CaiB) [reply Components/Inputs/TLV2544.cs]: Done.

File: Components/Inputs/TLV2544.cs (hunk 3 above)
Comment (CaiB) [reply Components/Inputs/TLV2544.cs]: This is a holdover from when I was planning on implementing auto-sampling, it has no utility when using single-shot, so I've removed it.

**Pull request merged**